src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -141,7 +141,11 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
                     }
                     else if (previousTask.IsCanceled)
                     {
-                        _state.TcsReadFromResponseStream.TrySetCanceled();
+                        _state.TcsReadFromResponseStream.TrySetCanceled(token);
+                    }
+                    else if (token.IsCancellationRequested)
```

Comment:
These two else blocks are identical and could be combined, but it's not a big deal.